src/hooks/useLatest.ts

```diff
@@ -0,0 +1,10 @@
+import { useRef, useEffect } from 'react';
+
+export function useLatest<T>(value: T) {
+  const ref = useRef<T>(value);
+  useEffect(() => {
```

Comment:
I wonder if using `useLayoutEffect` would a bit safer as it's the "sync" version of `useEffect`.

Reply:
I am not 100% sure. React docs uses `useEffect` in the example. Also these function are only called when user performs an action (apart from `selectCell` which can be called using gridRef). I think we can use `useEffect` for now and revisit if we find bugs.